packages/workbox-range-requests/src/lib/handle-range-request.js

```diff
@@ -0,0 +1,188 @@
+
+/**
+ * @memberof module:workbox-range-requests
+ *
+ * The public function `handleRangeRequest()` takes a `Request` and `Response`
+ * object as input, and returns one of two things:
+ *
+ * - a new `Response` with a status of `206 Partial Content`, containing a
+ * subset of the original `Response`, as determined by a valid `Range:` header
+ * provided in the `Request`.
+ * - a new `Response` with a status of `416 Range Not Satisfiable` if the
+ * subset of the original `Response` could not be determined for any reason
+ * given the inputs.
+ *
+ * `handleRangeRequest()` is intended to be used from standalone service worker
+ * code. If you want the same functionality and are already using `workbox-sw`
+ * or `workbox-runtime-caching`, then please see the examples of using the
+ * `CacheRangeResponsePlugin` class, which will be easier to use.
+ *
+ * @example
+ * self.addEventListener('fetch', (event) => {
+ *   if (event.request.headers.has('range')) {
+ *     event.respondWith(
+ *       // You need to ensure that your resource was previously cached, e.g.
+ *       // inside of an install event handler.
+ *       caches.match(event.request).then((cachedResponse) => {
+ *         return workbox.rangeRequests.handleRangeRequest({
+ *           request: event.request,
+ *           response: cachedResponse,
+ *         });
+ *       })
+ *     );
+ *   }
+ * });
+ */
+
+/**
+ * @private
+ * @param {Object} input
+ * @param {string} input.rangeHeader A Range: header value.
+ * @return {Object} An object with `start` and `end` properties, reflecting
+ * the parsed value of the Range: header. If either the `start` or `end` are
+ * omitted, then `null` will be returned.
+ */
+export function parseRangeHeader({rangeHeader} = {}) {
+  isType({rangeHeader}, 'string');
+
+  const normalizedRangeHeader = rangeHeader.trim().toLowerCase();
+  if (!normalizedRangeHeader.startsWith('bytes=')) {
+    throw ErrorFactory.createError('unit-must-be-bytes');
+  }
+
+  if (normalizedRangeHeader.includes(',')) {
+    throw ErrorFactory.createError('single-range-only');
+  }
+
+  const rangeParts = /(\d*)-(\d*)/.exec(normalizedRangeHeader);
+  // We need either at least one of the start or end values.
+  if (rangeParts === null || !(rangeParts[1] || rangeParts[2])) {
+    throw ErrorFactory.createError('invalid-range-values');
+  }
+
+  return {
+    start: rangeParts[1] === '' ? null : Number(rangeParts[1]),
+    end: rangeParts[2] === '' ? null : Number(rangeParts[2]),
+  };
+}
+
+/**
+ * @private
+ * @param {Object} input
+ * @param {Blob} input.blob A source blob.
+ * @param {Number|null} input.start The offset to use as the start of the
+ * slice.
+ * @param {Number|null} input.end The offset to use as the end of the slice.
+ * @return {Object} An object with `start` and `end` properties, reflecting
+ * the effective boundaries to use given the size of the blob.
+ */
+export function calculateEffectiveBoundaries({blob, start, end} = {}) {
```

Comment:
Since this method is exported publicly, should these be private? Otherwise it won't have docs.

If it is public, can you add a description of the function.